### PR TITLE
Install `rsync` before guest.pull()

### PR DESCRIPTION
**Pull request by thrix**
@mruprich reported that after running installability test the `tmt`
fails on missing rsync when pulling out artifacts.

After investiation, it turned out, installability for `rsync` removes
`rsync` from the test machine, but `tmt` still expects it to be around.

This tries to fix the problem and make sure `rsync` is still installed
on the machine before trying to pull out the artifacts.

Todo:
- [x] review comments
- [x] fix the issues with current tests
- [x] add a new test covering the problem

Signed-off-by: Miroslav Vadkerti <mvadkert@redhat.com>

### CODE REVIEW

**Comment by psss** [comment]
Not sure whether this check should go to the internal executor code. Should we include this rather under `pull()` and `push()` methods? We already have one similar code here:

https://github.com/psss/tmt/blob/4630cc7d573abb4bb814ae796b030fd91ccd8542/tmt/steps/prepare/__init__.py#L101

**Comment by thrix** [comment]
@psss ack, sounds like a better place, will move it there

File: tests/execute/rsync/test.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+rlJournalStart
+    rlPhaseStartTest "Test with rsync removed during execute"
+        rlRun "tmt run -avvvddd \
+            provision -h virtual \
```
Comment (psss) [inline tests/execute/rsync/test.sh:7]: Tests failed because of nested virtualization...

Comment (thrix) [reply tests/execute/rsync/test.sh:7]: yep, need to fix that

File: tmt/steps/provision/__init__.py [hunk 2]
```diff
@@ -462,6 +465,9 @@ def pull(self, source=None, destination=None, options=None):
             self.debug(f"Pull workdir from guest '{self.guest}'.")
         else:
             self.debug(f"Copy '{source}' from the guest to '{destination}'.")
+        # Make sure rsync is present, note: tests can remove it
+        self.debug('Ensure that rsync is installed on the guest.')
```
Comment (psss) [inline tmt/steps/provision/__init__.py:469]:
```suggestion
        self.debug('Ensure that rsync is installed on the guest.', level=3)
```

Move to level 3? Probably not that much interesting debug message.

File: tmt/steps/prepare/__init__.py [hunk 3]
```diff
@@ -69,11 +69,7 @@ def go(self):
             self.plan.provision.requires() +
```
Comment (psss) [inline tmt/steps/prepare/__init__.py:69]: We should probably also remove the `provision.requires()` part as it does not make sense now and causes enabling an extra `prepare` step which is not needed anymore. Or just modify the `Provision.requires()` method to return an empty list for now as there could possibly be other requires in the future?

**Review by psss — CHANGES_REQUESTED** [review]
Proposing a minor cleanup to prevent unnecessary `prepare` step with `rsync` only.

File: tmt/steps/provision/__init__.py [hunk 4]
```diff
@@ -429,6 +429,9 @@ def push(self, source=None, destination=None, options=None):
             self.debug(f"Push workdir to guest '{self.guest}'.")
         else:
             self.debug(f"Copy '{source}' to '{destination}' on the guest.")
+        # Make sure rsync is present, note: tests can remove it
+        self.debug('Ensure that rsync is installed on the guest.')
+        self.execute('rpm -q rsync || yum install -y rsync')
```
Comment (lukaszachy) [inline tmt/steps/provision/__init__.py:434]: Can we have helper function for this? It is duplicated code which harcodes rpm and yum, if we ever support non-rpm distros.

Also - is it OK to use yum? I'm not sure why we need that on rhel-8 and newer...

Comment (psss) [reply tmt/steps/provision/__init__.py:434]: The command is run on the guest so we need `yum` to make it working with both new and older distros.

File: tests/execute/rsync/test.sh (hunk 1 above)
Comment (thrix) [reply tests/execute/rsync/test.sh:7]: @psss just do not know how? what do you suggest?

File: tmt/steps/prepare/__init__.py (hunk 3 above)
Comment (thrix) [reply tmt/steps/prepare/__init__.py:69]: hmm, not sure I follow @psss, it now returns an empty list?

Comment (thrix) [reply tmt/steps/prepare/__init__.py:69]: ah found it, I made it return `[]`

File: tmt/steps/provision/__init__.py (hunk 4 above)
Comment (thrix) [reply tmt/steps/provision/__init__.py]: added a helper function to `Guest` class

File: tests/execute/rsync/test.sh (hunk 1 above)
Comment (thrix) [reply tests/execute/rsync/test.sh:7]: adjust :) done :)

**Comment by thrix** [comment]
we  can try again the review @psss @lukaszachy

**Comment by psss** [comment]
Looks like there is a syntax error:

    Sorry: IndentationError: expected an indented block (__init__.py, line 18)

@thrix, please, have a look.

**Comment by thrix** [comment]
wtf, what does this autopep8 module do? :D

File: tests/execute/rsync/test.sh (hunk 1 above)
Comment (psss) [reply tests/execute/rsync/test.sh:12]: The test ran the whole `tmt` test suite. Added tmp directory to run only the `rsync` removal command.

**Review by psss — APPROVED** [review]
Thanks for fixing this. Added a couple of adjustments in bc4ae18. Please, have a look if everything's fine.

**Comment by thrix** [comment]
LGTM, let's hit it

**Pull request merged**